### PR TITLE
Add vesting transfers support & display token details

### PR DESCRIPTION
**Pull request by stefashkaa**
### **Description**
- **What does this PR do?**
  - _Added an ability to send tokens with vesting option_
  - _Added an ability to view token details (not only XOR like it was)_
  
- **Why is this change needed?**
  - _To allow users making vested transfers_
  - _To display token details with locked balance for receivers_

### **Type of change**
- [ ] Bug fix
- [x] New feature
- [ ] Refactor (cleaning up code without changing functionality)
- [ ] Documentation update
- [ ] Tests
- [ ] Other (describe below):

---

### **Checklist**
- [x] Code adheres to coding guidelines and standards.
- [x] Linting and formatting checks have passed, no local issues.
- [x] Existing tests pass successfully, no local errors.
- [x] Pull request is linked to a relevant issue or task.
- [ ] Tests are written for new features or fixes.
- [ ] Documentation has been updated (if applicable).
  
---

### **Testing**

- **How has this been tested?**
  - Proceed with the main vested transfer flow: enable vesting, input all params of the TX and submit it
- **Edge cases covered:**
  - Try to proceed without vesting percentage
  - Try to switch on/off vesting and check fees difference
  - Since fee depends on the different params (percentage/vesting switched on/off) there might be issues in case user doesn't have enough fees to proceed with the TX
---

### **Related Issue/Task**
- _[Add vesting](https://github.com/sora-xor/polkaswap-exchange-web/issues/1545)_ 
- _[Show locked tokens](https://github.com/sora-xor/polkaswap-exchange-web/issues/1513)_

---

### **Screenshots/Video/Link to dedicated environment (if applicable)**

image

image

---

### **Reviewer Tasks**
- [ ] Review logic for correctness and clarity.
- [ ] Verify that the code is following best practices.
- [ ] Test or review related areas if this PR affects other projects.

### CODE REVIEW

File: src/store/account/actions.ts [hunk 1]
```diff
@@ -346,6 +348,37 @@ const actions = defineActions({
...
+    { to, asset, amount, vestingPercent, unlockPeriodInDays }: VestedTransferParams
+  ): Promise<void> {
+    const {
+      rootState: {
+        wallet: {
+          settings: { blockNumber },
+        },
+      },
+    } = rootActionContext(context);
+    await api.assets.vestedTransfer(asset, to, amount, blockNumber, vestingPercent, unlockPeriodInDays);
```
Comment (VladislavPopovSR) [inline src/store/account/actions.ts:380]: Add try catch please

File: src/components/WalletSend.vue [hunk 2]
```diff
@@ -165,6 +188,13 @@ export default class WalletSend extends Mixins(
   NetworkFeeWarningMixin
 ) {
   readonly delimiters = FPNumber.DELIMITERS_CONFIG;
+  readonly vestingPeriods = [
+    { value: 1, label: '1 day' },
+    { value: 7, label: '7 days' },
+    { value: 30, label: '30 days' },
+    { value: 60, label: '60 days' },
+    { value: 90, label: '90 days' },
+  ];
```
Comment (VladislavPopovSR) [inline src/components/WalletSend.vue]: Label i18n?

File: src/store/account/actions.ts (hunk 1 above)
Comment (stefashkaa) [reply src/store/account/actions.ts:380]: Will be handled in a method which uses it

File: src/components/WalletSend.vue (hunk 2 above)
Comment (VladislavPopovSR) [reply src/components/WalletSend.vue]: Marked as to do :)

**Review by VladislavPopovSR — APPROVED** [review]

**Pull request merged**